src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,323 @@
+        await this.updateUsers(groupId!, 'owners', ownerIds);
+      }
+      else if (this.verbose) {
+        await logger.logToStderr(`No owners to update.`);
+      }
+
+      memberIds = await this.getUserIds(logger, args.options.memberIds, args.options.memberUserNames);
+      if (memberIds.length !== 0) {
+        await this.updateUsers(groupId!, 'members', memberIds);
+      }
+      else if (this.verbose) {
+        await logger.logToStderr(`No members to update.`);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  };
+
+  private createRequestBody(options: Options): any {
+    const requestBody: any = {
+      displayName: options.newDisplayName,
+      description: options.description,
+      mailNickName: options.mailNickname,
+      visibility: options.visibility
+    };
+
+    this.addUnknownOptionsToPayload(requestBody, options);
+    return requestBody;
+  }
+
+  private async getUserIds(logger: Logger, userIds?: string, userNames?: string): Promise<string[]> {
+    if (userIds) {
+      return formatting.splitAndTrim(userIds);
+    }
+
+    if (userNames) {
+      return entraUser.getUserIdsByUpns(formatting.splitAndTrim(userNames));
+    }
+
+    return [];
+  }
+
+  private async updateUsers(groupId: string, role: string, userIds: string[]): Promise<void> {
+    const groupUsers = await odata.getAllItems<User>(`${this.resource}/v1.0/groups/${groupId}/${role}/microsoft.graph.user?$select=id`);
+    const userIdsToAdd = userIds.filter(userId => groupUsers.findIndex(groupUser => groupUser.id === userId) === -1);
+    const userIdsToRemove = groupUsers.filter(groupUser => userIds.findIndex(userId => groupUser.id === userId) === -1);
+
+    for (let i = 0; i < userIdsToAdd.length; i += 400) {
+      const userIdsBatch = userIdsToAdd.slice(i, i + 400);
+      const requestOptions = this.getRequestOptions();
+
+      // only 20 requests per one batch are allowed
+      for (let j = 0; j < userIdsBatch.length; j += 20) {
+        // only 20 users can be added in one request
+        const userIdsChunk = userIdsBatch.slice(j, j + 20);
+        requestOptions.data.requests.push({
+          id: j + 1,
+          method: 'PATCH',
+          url: `/groups/${groupId}`,
+          headers: {
+            'content-type': 'application/json;odata.metadata=none'
+          },
+          body: {
+            [`${role}@odata.bind`]: userIdsChunk.map(u => `${this.resource}/v1.0/directoryObjects/${u}`)
+          }
+        });
+      }
+
+      const res = await request.post<{ responses: { status: number; body: any }[] }>(requestOptions);
+      for (const response of res.responses) {
+        if (response.status !== 204) {
+          throw response.body;
+        }
+      }
+    }
+
+    for (let i = 0; i < userIdsToRemove.length; i += 20) {
+      const userIdsBatch = userIdsToRemove.slice(i, i + 20);
+      const requestOptions = this.getRequestOptions();
+
+      userIdsBatch.map(userId => {
```

Comment:
Seems like there's something wrong when removing users.

![image](https://github.com/user-attachments/assets/cf508f52-084a-4b55-ace5-1439a9f06c48)

Seems like the ID passed in the batch request is incorrect.

Reply:
fixed